sys/include/net/gnrc/netif/internal.h

```diff
@@ -466,9 +454,34 @@ int gnrc_netif_ipv6_iid_from_addr(const gnrc_netif_t *netif,
  */
 int gnrc_netif_ipv6_iid_to_addr(const gnrc_netif_t *netif, const eui64_t *iid,
                                 uint8_t *addr);
+
+/**
+ * @brief   Converts an interface IID of an interface's hardware address
+ *
+ * @param[in] netif     The network interface @p iid came from
+ * @param[out] iid      The IID based on gnrc_netif_t::device_type
+ *
+ * @return  `sizeof(eui64_t)` on success.
+ * @return  -ENOTSUP, if interface has no link-layer address or if
+ *          gnrc_netif_t::device_type is not supported.
+ */
+static inline int gnrc_netif_ipv6_get_iid(gnrc_netif_t *netif, eui64_t *iid)
+{
+#if GNRC_NETIF_L2ADDR_MAXLEN > 0
+    if (netif->flags & GNRC_NETIF_FLAGS_HAS_L2ADDR) {
+        return gnrc_netif_ipv6_iid_from_addr(netif,
```

Comment:
This can also return `-EINVAL` [1]. Maybe you add a hint to the doc of `gnrc_netif_ipv6_iid_from_addr()`?

[1] https://github.com/RIOT-OS/RIOT/blob/master/sys/net/gnrc/netif/gnrc_netif_device_type.c#L38

Reply:
Down